lib/internal/util/comparisons.js

```diff
@@ -117,42 +99,36 @@ function isEqualBoxedPrimitive(val1, val2) {
 // (e.g. a Uint8Array and a Uint16Array with the same memory content
 // could still be different because they will be interpreted differently).
 //
-// For strict comparison, objects should have
+// To be treated as deep equal, objects should have:
 // a) The same built-in type tags
 // b) The same prototypes.
 
-function innerDeepEqual(val1, val2, strict, memos) {
+function isDeepStrictEqual(val1, val2, options = {}) {
+  const { memos } = options;
```

Comment:
Keeping the former notation would cause less churn at the moment and it would also allow to skip the prototype check in the non strict equality case. It is also likely faster.

Reply:
The former way of passing through `strict` is indeed faster than using the options object. See https://ci.nodejs.org/view/Node.js%20benchmark/job/benchmark-node-micro-benchmarks/401/console. The regressions in the strict mode come from this.

Reply:
You had me at the "less churn" argument, but yeah, "more performant" is even more convincing.

Reply:
OK, de-churned.